v2/internal/runner/runner_test.go

```diff
@@ -24,3 +31,185 @@ func Test_createReportingOptions(t *testing.T) {
+		HttpClient: &retryablehttp.Client{
+			HTTPClient: &http.Client{
+				Transport: &http.Transport{
+					ProxyConnectHeader: header,
+				},
+			},
+		},
+	}
+	err := yamlwrapper.DecodeAndValidate(strings.NewReader(data), reportingOptions)
+	require.Nil(t, err)
+
+	os.Setenv("GITHUB_USER", "testuser")
+	os.Setenv("GITHUB_TOKEN", "tokentesthere")
+	os.Setenv("GITHUB_PROJECT", "testproject")
+
+	assignEnvVarToReportingOpt(reportingOptions)
+	assert.Equal(t, "testuser", reportingOptions.GitHub.Username)
+	assert.Equal(t, "tokentesthere", reportingOptions.GitHub.Token)
+	assert.Equal(t, "testproject", reportingOptions.GitHub.ProjectName)
+}
+
+func Test_assignEnvVarToReportingOptEmptyField(t *testing.T) {
+	data := `
+github:
+  username: ""
+  owner: $GITHUB_OWNER
+  token: $GITHUB_TOKEN
+  project-name: $GITHUB_PROJECT
+  issue-label: $ISSUE_LABEL
+  severity-as-label: false`
+
+	header := http.Header{}
+	header.Add("test", "test")
+
+	reportingOptions := &reporting.Options{
+		HttpClient: &retryablehttp.Client{
+			HTTPClient: &http.Client{
+				Transport: &http.Transport{
+					ProxyConnectHeader: header,
+				},
+			},
+		},
+	}
+	err := yamlwrapper.DecodeAndValidate(strings.NewReader(data), reportingOptions)
+	require.NotNil(t, err)
+}
+
+func Test_assignEnvVarToReportingOptFailed(t *testing.T) {
+	data := `
+github:
+  username: $GITHUB_USER
+  owner: $GITHUB_OWNER
+  token: $GITHUB_TOKEN
+  project-name: $GITHUB_PROJECT
+  issue-label: $ISSUE_LABEL
+  severity-as-label: false`
+
+	header := http.Header{}
+	header.Add("test", "test")
+
+	reportingOptions := &reporting.Options{
+		HttpClient: &retryablehttp.Client{
+			HTTPClient: &http.Client{
+				Transport: &http.Transport{
+					ProxyConnectHeader: header,
+				},
+			},
+		},
+	}
+	err := yamlwrapper.DecodeAndValidate(strings.NewReader(data), reportingOptions)
+	require.Nil(t, err)
+
+	os.Setenv("GITHUB_USER", "testuser")
+
+	assignEnvVarToReportingOpt(reportingOptions)
+	assert.NotEqual(t, "$GITHUB_USER", reportingOptions.GitHub.Username)
+}
+
+func Test_assignEnvVarToReportingOptFailedMultiple(t *testing.T) {
+	data := `
+github:
+  username: $GITHUB_USER
+  owner: $GITHUB_OWNER
+  token: $GITHUB_TOKEN
+  project-name: $GITHUB_PROJECT
+  issue-label: $ISSUE_LABEL
+  severity-as-label: false`
+
+	header := http.Header{}
+	header.Add("test", "test")
+
+	reportingOptions := &reporting.Options{
+		HttpClient: &retryablehttp.Client{
+			HTTPClient: &http.Client{
+				Transport: &http.Transport{
+					ProxyConnectHeader: header,
+				},
+			},
+		},
+	}
+	err := yamlwrapper.DecodeAndValidate(strings.NewReader(data), reportingOptions)
+	require.Nil(t, err)
+
+	os.Setenv("GITHUB_USER", "testuser")
+	os.Setenv("GITHUB_PROJECT", "testproject")
+
+	assignEnvVarToReportingOpt(reportingOptions)
+	assert.Equal(t, "testuser", reportingOptions.GitHub.Username)
+	assert.NotEqual(t, "$GITHUB_PROJECT", reportingOptions.GitHub.ProjectName)
+}
+
+type TestStruct1 struct {
```

Comment:
The previous tests can be rewritten more concisely using generic shorter structs, as we did here. The initial example involving YAML unmarshaling was provided only to pinpoint the issue with untagged YAML fields and when the replacement logic was supposed to be implemented, implementing the YAML unmarshal interface for the specific struct of type `reporting.Options`